src/main/java/eu/deltasource/internship/abankingsystem/model/Transaction.java

```diff
@@ -0,0 +1,176 @@
+package eu.deltasource.internship.abankingsystem.model;
+
+import eu.deltasource.internship.abankingsystem.enums.Currency;
+
+import java.time.LocalDate;
+import java.util.Objects;
+
+public class Transaction {
+
+    private int id;
+
+    private final BankAccount sourceAccount;
+
+    private final BankAccount targetAccount;
```

Comment:
If the target account is in another bank you wouldn't have access to it so i think it is more appropriate to use iban even for the source account

Reply:
Done. Still, please if you can, check if I understood you correctly.